posts/Guillaume/2017-03-30-hibernate-validator-600-alpha2-out.adoc

```diff
@@ -0,0 +1,131 @@
+                .constraint( new SizeDef().min( 3 ).max( 10 ) )
+            .containerElementType( 1 )
+                .constraint( new MinDef().value( 1 ) );
+----
+
+or via an XML mapping configuration (which is part of the Bean Validation 2.0 specification):
+
+[source, xml]
+----
+[...]
+    <bean class="com.acme.FishTank" ignore-annotations="false">
+        <field name="model">
+            <containerElementType>
+                <valid />
+            </containerElementType>
+        </field>
+        <field name="fishCountByType">
+            <containerElementType typeArgumentIndex="0">
+                <constraint annotation="javax.validation.constraints.Size">
+                    <element name="min">
+                        <value>3</value>
+                    </element>
+                    <element name="max">
+                        <value>10</value>
+                    </element>
+                </constraint>
+            </containerElementType>
+            <containerElementType typeArgumentIndex="1">
+                <constraint annotation="javax.validation.constraints.Min">
+                    <element name="value">
+                        <value>1</value>
+                    </element>
+                </constraint>
+            </containerElementType>
+        </field>
+    </bean>
+[...]
+----
+
+==== Other improvements
+
+We also made a couple of more minor improvements:
+
+ * the `TYPE_ARGUMENT` node type has been renamed to `CONTAINER_ELEMENT` (type argument don't make sense for annotated array elements)
+ * we allow per context custom value extractors
+ * we allow value extractors to be contributed using the service loader mechanism
+ * we allow to specify value extractors using XML configuration
+ * we explore the parent classes and interfaces to search the `ValueExtractor` definition
+
+=== Support for the new constraints added to Bean Validation 2.0
+
+In the Alpha2 release of Bean Validation 2.0, we introduced the following new constraints:
+
+ * `@NotBlank`: check that a char sequence is not blank (ie not null, and length of trimmed char sequence > 0)
+ * `@NotEmpty`: check that an element (char sequence, collection, array) is not null and not empty
+ * `@Email`: check that a char sequence is a valid email
+ * `@Positive`: check that a number is positive
+ * `@Negative`: check that a number is negative
+
+We added support for these new constraints to Hibernate Validator.
+
+=== Performance improvements
+
+We started to run a few benchmarks on Hibernate Validator 6 and we fixed a few performance regressions.
+
+6.0.0.Alpha2 is now significantly faster than 5.4, mostly thanks to the new `ValueExtractor` design.
+
+== Getting 6.0.0.Alpha2
+
+To get the release with Maven, Gradle etc. use the GAV coordinates _org.hibernate.validator:{hibernate-validator|hibernate-validator-cdi|hibernate-validator-annotation-processor}:6.0.0.Alpha2_. Note that the group id has been changed to `org.hibernate.validator`.
```

Comment:
Someone might think the group id has changed *again*. Perhaps something like this:

> Note that the group id has changed from `org.hibernate` (Hibernate Validator 5 and earlier) to `org.hibernate.validator` (from Hibernate Validator 6 onwards).